circleci/images/exttester/Dockerfile

```diff
@@ -131,9 +131,15 @@ RUN echo "deb https://apt.postgresql.org/pub/repos/apt bullseye-pgdg main" >> /e
  && apt-get install -y --no-install-recommends \
     postgresql-client-${PG_MAJOR}=${pgdg_version} \
     postgresql-${PG_MAJOR}=${pgdg_version} \
-    postgresql-server-dev-${PG_MAJOR}=${pgdg_version} \
-# clear apt cache
- && rm -rf /var/lib/apt/lists/*
+    postgresql-server-dev-${PG_MAJOR}=${pgdg_version}
+
+# All pg versions are dependent upon the latest libpq5 version which causes runtime errors
+# The following block removes installed libpq5 and install the libpq5 version related to postgres version
+RUN export LIBPQ_VERSION=${PG_VERSION}-1 \
+    && apt-get remove -y libpq5 \
+    && apt-get install --allow-downgrades -y libpq5=${LIBPQ_VERSION}.pgdg110+1 \
```

Comment:
will this always be `${PG_VERSION}-1.pgdg110+1`? it seems to me the last part is very release specific.

The same I have observed for the postgres versions uses. A simple smoke test could be to make some of the older images, like `12.6` and `11.3` or sth.

The versions being different being the reasoning fro the more complexities above, especially on line 130, where we `grep | awk | head` to find the actual `pgdg_version` fro a given `${PG_VERSION}`.

Reply:
Yes as you can see from the pgdg repo list below this has the same pattern
[libpq_versions.txt](https://github.com/citusdata/the-process/files/7667322/libpq_versions.txt)

Reply:
I used pgdg_version parameter to define libpq5 versions

Reply:
what is the difference between the following two
```
libpq5/bullseye-pgdg-archive 13.4-4.pgdg110+1 amd64
libpq5/bullseye-pgdg-archive 13.4-1.pgdg110+1 amd64
```
Would it be ok to install `-1.pgdg110+1`?
If security problems are found they sometimes pull old releases and release new ones like the `-4`.

We can simply ignore it for now and cross that bridge when we get there.

Reply:
The last parameter -1 and -4 and fancy version which is a fix on the minor version. If there are issues on minor version then a new fancy version is released. For us it is important to download the latest one so in this case we need to download 13.4-4
The command below already addresses this issue
https://github.com/citusdata/the-process/blob/da58cb4c95b585931a5afa4bacaef972fd7b22a2/circleci/images/exttester/Dockerfile#L130
The images below are test results from the exttester image
![image](https://user-images.githubusercontent.com/1596331/145172557-3dc4a14b-891b-441b-ab73-9f234f75ed35.png)

Reply:
Cool, this change makes sense to me.

Could you make sure we install the same libpq versions in our other containers:
 - https://github.com/citusdata/the-process/blob/master/circleci/images/failtester/Dockerfile#L59
 - https://github.com/citusdata/the-process/blob/master/circleci/images/extbuilder/Dockerfile#L55 (although I don't think we use it much here, might still be good to have the same version during building and execution.)
  - https://github.com/citusdata/the-process/blob/master/circleci/images/citusupgradetester/Dockerfile#L60
  - https://github.com/citusdata/the-process/blob/master/circleci/images/pgupgradetester/Dockerfile#L46-L62 (this is a harder one as it installs multiple postgres versions in 1 container to test pg upgrades)

Reply:
I think it is better to build the project using the libpq compatible with the postgres version. Therefore I will update the libpq versions for exbuilder

Reply:
citusupgradeversion depends on multiple postgres versions as well. I got error while getting postgres version in case of libpq versionupdate as following
https://github.com/citusdata/the-process/runs/4503115973?check_suite_focus=true#step:4:1359

Reply:
https://github.com/citusdata/the-process/runs/4503115973?check_suite_focus=true#step:4:1085

The way you drop loibpq5 removes `postgresql-12`, could you try instead of dropping `libpq5` install the right version when we install the other packages by including `libpq5=${pgdg_version}` in the command there?

Reply:
Removed apt-get remove and only used install with downgrade option
I got error for cituspgupgradetester
https://github.com/citusdata/the-process/runs/4506524481?check_suite_focus=true
I can not change the version of pgupgradetester since only one libpq installation allowed

Reply:
if only 1 version is allowed I think we should go for the latest version available, in our case the one that comes with pg14.

Reply:
I had problems for failtester since some tests failed. I can not identify the problem. @thanodnl Should I rollback for this image?
Tests
https://app.circleci.com/pipelines/github/citusdata/citus/19914/workflows/1417abc8-e8e2-4d25-8d55-e6f610998d97

Reply:
Quicly looking I think there are 2 classes of errors:
 - flappy tests
 - output changed due to libpq versions
 
 for the latter, an example is this test https://app.circleci.com/pipelines/github/citusdata/citus/19914/workflows/1417abc8-e8e2-4d25-8d55-e6f610998d97/jobs/480078?invite=true#step-107-21
 
 You see that the libpq version we had installed before (latest) had a pretty big output, while the older version we currently install in the image has only a sinlge line. This is to be expected for these changes and need to be reflected on the Citus repo.
 
 For the first class, I suspect this test to be flappy: https://app.circleci.com/pipelines/github/citusdata/citus/19914/workflows/1417abc8-e8e2-4d25-8d55-e6f610998d97/jobs/480076
 For this I have initiated a rerun from failed.
 
 Could you refrain in the future from reverting changes when you run into failures? It is really hard to help you if the latest versions do not reflect errors you run into.

Reply:
The rerun from failed keeps failing on the arbitrary failure tests. In that case it is good to work with the author of the tests to understand why they are failing. To find the author of a test you need to look at the files that has the failure; `expected/ch_benchmarks_1.out`: https://github.com/citusdata/citus/blame/master/src/test/regress/expected/ch_benchmarks_1.out, which is Talha. Since he is not really around anymore you could consult Onder. I am not completely sure what this test is doing, and why a libpq version change seems to trigger problems with Connection limits being exceeded (also a good reason to reach out to Onder as he has most context around that area)